assets/js/amp-customize-controls.js

```diff
@@ -0,0 +1,194 @@
+		 */
+		function toggleAmpView() {
+			api.previewer.previewUrl.set( setCurrentAmpUrl( api.previewer.previewUrl.get() ) );
+		}
+
+		/**
+		 * Inspect iFrame metadata to see if loaded page was successfully AMP'd.
+		 *
+		 * @return {void}
+		 */
+		function isPageAmpified() {
+			var ampEnabled = false,
+				headContent = $( api.previewer.preview.iframe ).contents().find( 'head' ).html();
+
+			if ( headContent.search( '<meta name="generator" content="AMP Plugin' ) >= 0 ) {
+				ampEnabled = true;
+			}
+
+			// Update tooltip display based on above IF it's currently toggled on.
+			if ( ampToggle && ! ampEnabled ) {
+				toggleControl( false, true, true );
+			} else if ( ! ampToggle && ! ampEnabled ) {
+
+				// Enable toggle but don't trigger anything else.
+				$( '.amp-toggle input' ).prop( 'disabled', false );
+			}
+		}
+
+		/**
+		 * Central function to control UI toggle elements.
+		 *
+		 * @param {bool} checked If toggle input should be checked.
+		 * @param {bool} disabled If toggle input should be disabled.
+		 * @param {bool} notification If tooltip notification should appear.
+		 * @return {void}
+		 */
+		function toggleControl( checked, disabled, notification ) {
+			var $input = $( '.amp-toggle input' ),
+				$tooltip = $( '.amp-toggle .tooltip' ),
+				tooltipTimer = 5000;
+
+			$input.prop( 'checked', checked );
+			$input.prop( 'disabled', disabled );
+
+			if ( notification ) {
+				$tooltip.fadeIn();
+				setTimeout( function() {
+					$tooltip.fadeOut();
+				}, tooltipTimer );
+			} else {
+				$tooltip.hide();
+			}
+
+			$input.trigger( 'change' );
+		}
+
+		// AMP panel triggers the input toggle for AMP preview.
+		panel.expanded.bind( function() {
+			var panelOpen = panel.expanded.get();
+			toggleControl( panelOpen, panelOpen, false );
+		} );
+
+		// If AMP panel is open, let it handle enable/disable of the toggle,
+		// otherwise auto enable toggle when mobile device is selected.
+		api.previewedDevice.bind( function( device ) {
+			if ( ! panel.expanded.get() && ! ampToggle ) {
+				toggleControl( 'mobile' === device, false, false );
+			}
+		} );
+
+		// Preview is ready, check if it's AMP'd in case we need to show tooltip.
+		api.previewer.bind( 'ready', isPageAmpified );
+
+		// Persist the presence or lack of the amp=1 param when navigating in the preview,
+		// even if current page is not yet supported.
+		api.previewer.previewUrl.validate = ( function( prevValidate ) {
+			return function( value ) {
+				var val = prevValidate.call( this, value );
+				if ( val ) {
+					val = setCurrentAmpUrl( val );
+				}
+				return val;
+			};
+		} )( api.previewer.previewUrl.validate );
+
+		// Adding checkbox toggle before device selection.
+		$( '.devices-wrapper' ).before( '<label class="amp-toggle">' +
+			'<input type="checkbox">' +
+			'<span class="slider"></span>' +
+		'</label>' );
+
+		// Notification tooltip for AMP toggle.
+		$( '.amp-toggle input' ).before( '<span class="tooltip">' +
```

Comment:
We should to use `wp.template` here instead of string concatenation. If the template is output via PHP we can also then make use of i18n in PHP to get the translated string.